pkg/reconciler/reconciler.go

```diff
@@ -609,6 +622,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 		}
 	}()
 
+	if r.pauseReconcileAnnotation != "" {
+		if v, ok := obj.GetAnnotations()[r.pauseReconcileAnnotation]; ok {
+			if v == "true" {
+				log.Info(fmt.Sprintf("Resource has '%s' annotation set to 'true', reconcile stopped", r.pauseReconcileAnnotation))
+				u.UpdateStatus(
+					updater.EnsureCondition(conditions.Paused(corev1.ConditionTrue, conditions.ReasonPauseReconcileAnnotationSet, "")))
+				return ctrl.Result{}, nil
+			}
+		}
+	}
+
+	u.UpdateStatus(
+		updater.EnsureCondition(conditions.Paused(corev1.ConditionFalse, "", "")))
```

Comment:
```suggestion
                // TODO(porridge): change to updater.EnsureCondition(conditions.Paused(corev1.ConditionFalse, "", "")))
                // once stackrox operator with pause support is released.
                // At that time also add `Paused` to the list of conditions expected in stackrox operator e2e tests.
                // Otherwise, the number of conditions in the `status.conditions` list will vary depending on the version
                // of used operator, which is cumbersome due to https://github.com/kudobuilder/kuttl/issues/76
		updater.RemoveCondition(conditions.TypePaused)
```
While